distributed/shuffle/_arrow.py

```diff
@@ -45,18 +45,12 @@ def check_minimal_arrow_version() -> None:
         )
 
 
-def convert_partition(data: bytes, meta: pd.DataFrame) -> pd.DataFrame:
+def convert_shards(shards: list[pa.Table], meta: pd.DataFrame) -> pd.DataFrame:
```

Comment:
(disclaimer: still in early review) I once tried to move tables around instead of bytes but that messed up the event loop. We should check this before merging